src/index.ts

```diff
@@ -149,7 +149,15 @@ export function register (options: Options = {}): () => Register {
   const shouldCache = !!(options.cache == null ? DEFAULTS.cache : options.cache)
   const fast = !!(options.fast == null ? DEFAULTS.fast : options.fast)
   const project = options.project || DEFAULTS.project
-  const cacheDirectory = options.cacheDirectory || DEFAULTS.cacheDirectory || join(tmpdir(), 'ts-node')
+
+  function getTmpDir(): string {
```

Comment:
Can you move this function outside of the main run loop and add a space after the function name to the parenthesis to match the rest of the project? Also, I'll double check to make sure I didn't miss it, but why base it on the home directory instead of username or cwd?